src/nss/new_nss_oslogin.c

```diff
@@ -241,3 +241,174 @@ int recvline(struct Buffer *const buffer) {
 
   return -1;  // Unreachable code.
 }
+
+static enum nss_status
+_nss_oslogin_getpwnam_r(const char *name, struct passwd *result, char *buffer,
+                        size_t buflen, int *errnop) {
+  int res;
+  struct Buffer mgr;
+  memset(&mgr, 0, sizeof(struct Buffer));
+
+  *errnop = 0;
+  mgr.bufsize = BUFSIZE;
+  mgr.buf = (char *)malloc(BUFSIZE);
```

Comment:
Is this module only planned to be used only in short running processes? 

I am worried about memory leaks if this module is used in the future by Daemons or other long running tasks. There was a similar pattern in the previous CL. I just want to make sure we're not going to cause problems in the future with memory management if someone uses this module for other purposes.

Reply:
this comment confused me, bc this function returns and has nothing short or long about it. but i realized that you are talking about some missing calls to free(). updated PR accordingly